File: BFS-DFS/43162.py
def DFS(graph, v, visited):
    visited[v] = True
    n = len(graph)
    
    for node in range(n):
        if graph[v][node] == 1 and visited[node] == False:
            DFS(graph, node, visited)
    

def solution(n, computers):
    answer = 0
    visited = n * [0]    
    for start in range(n):
        if visited[start] == 0:
            DFS(computers, start, visited)
            answer += 1
            
    return answer


solution(3, [[1,1,0], [1,1,0], [0,0,1]])

File: Javascript/BFS-DFS/43162.js
function solution(n, computers) {
    let answer = 0;
    let visited = [];
    
    for (let i = 0; i < n; i++){
        visited.push(false);
    }
    
    var DFS = function(computers, i) {
        if(visited[i]) return; 
        visited[i] = true;

        for(var j = 0; j < computers.length; j++) {
            if(computers[i][j] === 1) {
                console.log(i + ', ' + j);
                console.log('connected');
                DFS(computers, j);
            }
        }
    }
    
    
    for(var i = 0; i < n; i++) {
        if(!visited[i]) {
            answer++;
            console.log(answer);
            DFS(computers, i);
        }
    }
    
    return answer;
}